src/uct/ib/ud/base/ud_iface.c

```diff
@@ -693,9 +731,11 @@ ucs_status_t uct_ud_iface_query(uct_ud_iface_t *iface,
                                          UCT_IFACE_FLAG_CB_SYNC          |
                                          UCT_IFACE_FLAG_CB_ASYNC         |
                                          UCT_IFACE_FLAG_ERRHANDLE_PEER_FAILURE;
-    iface_attr->cap.event_flags        = UCT_IFACE_FLAG_EVENT_SEND_COMP |
-                                         UCT_IFACE_FLAG_EVENT_RECV      |
-                                         UCT_IFACE_FLAG_EVENT_ASYNC_CB;
+    if (uct_ib_iface_device(&iface->super)->req_notify_cq_support) {
+        iface_attr->cap.event_flags = UCT_IFACE_FLAG_EVENT_SEND_COMP |
+                                      UCT_IFACE_FLAG_EVENT_RECV |
+                                      UCT_IFACE_FLAG_EVENT_ASYNC_CB;
+    }
```

Comment:
we need to disable AM_SHORT somehow if `msx_sge` is not enough.. Maybe disable it unconditionally for efa?

Reply:
as discussed, will evaluate supporting AM_SHORT with tweaked `config.am.max_short = sizeof(uct_ud_neth_t) + 1` to force max sge 2.

Reply:
tentatively fixed

Reply:
pushed disabling AM_SHORT, initially tested with perftest, now checking all the gtests.